src/particle_restart.cpp

```diff
@@ -93,33 +94,27 @@ void run_particle_restart()
   int64_t particle_seed;
   switch (previous_run_mode) {
   case RunMode::EIGENVALUE:
-    particle_seed = (simulation::total_gen + overall_generation() - 1)*settings::n_particles + p.id_;
+    particle_seed = (simulation::total_gen + overall_generation() - 1) *
+                      settings::n_particles +
+                    p.id();
     break;
   case RunMode::FIXED_SOURCE:
-    particle_seed = p.id_;
+    particle_seed = p.id();
     break;
   default:
     throw std::runtime_error{"Unexpected run mode: " +
       std::to_string(static_cast<int>(previous_run_mode))};
   }
-  init_particle_seeds(particle_seed, p.seeds_);
+  init_particle_seeds(particle_seed, p.seeds());
 
   // Force calculation of cross-sections by setting last energy to zero
   if (settings::run_CE) {
-    for (auto& micro : p.neutron_xs_) micro.last_E = 0.0;
+    p.invalidate_neutron_xs();
   }
 
   // Prepare to write out particle track.
-  if (p.write_track_) add_particle_track(p);
-
-  // Every particle starts with no accumulated flux derivative.
```

Comment:
Ah, I see what you're talking about. I put this stuff in the `ParticleData` constructor. It's because the logic for this looks different between the SOA and non-SOA particle. That way we keep the common stuff outside the particle class, and the uncommon stuff restricted to the `ParticleData` class.

Sorry to not include an explainer on this. You're right that doing multiple smaller PRs would have made documenting changes like this easier. Will definitely do that in the future.